### PR TITLE
EventHub Source and Sink

### PR DESCRIPTION
**Pull request by jo-rivera**
Implementation of EventHub source and sink blocks
ADO#7660

### CODE REVIEW

**Review by bquantump — APPROVED** [review]

**Review by pomeroy3 — COMMENTED** [review]
Just some general questions

File: gr-azure-software-radio/python/eventhub_sink.py
Comment (bquantump) [inline gr-azure-software-radio/python/eventhub_sink.py:76]: any downside to using batch here? Batch is typically meant for N > 1 messages.

I think may just want to send event.

File: gr-azure-software-radio/python/eventhub_sink.py
Comment (bquantump) [inline gr-azure-software-radio/python/eventhub_sink.py:106]: Shouldn't this integrate with Marko's default creds stuff?

File: gr-azure-software-radio/python/eventhub_source.py
Comment (bquantump) [reply gr-azure-software-radio/python/eventhub_source.py:94]: I think we may be fine since @jo-rivera in not using multiprocess threading, so should die with the this process. 

we should confirm though @jo-rivera

File: gr-azure-software-radio/python/eventhub_sink.py
Comment (jo-rivera) [reply gr-azure-software-radio/python/eventhub_sink.py:76]: API for the PublisherClient only supports "send_batch" https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/latest/azure.eventhub.html

File: gr-azure-software-radio/python/eventhub_sink.py
Comment (jo-rivera) [reply gr-azure-software-radio/python/eventhub_sink.py:106]: I created user story ADO#8095 to address this. The Blob blocks can also benefit from the DefaultCredentials block.

File: gr-azure-software-radio/grc/azure_software_radio_eventhub_sink.block.yml
Comment (jo-rivera) [reply gr-azure-software-radio/grc/azure_software_radio_eventhub_sink.block.yml]: the default is empty string which sets partition_id to None and the client sends to all partitions in this case

File: gr-azure-software-radio/grc/azure_software_radio_eventhub_source.block.yml
Comment (jo-rivera) [reply gr-azure-software-radio/grc/azure_software_radio_eventhub_source.block.yml]: set to @latest

File: gr-azure-software-radio/python/eventhub_source.py
Comment (jo-rivera) [reply gr-azure-software-radio/python/eventhub_source.py:98]: it does but is good practice to stop the thread. I updated the code.

File: gr-azure-software-radio/examples/README.md
Comment (pomeroy3) [inline gr-azure-software-radio/examples/README.md:95]: I couldn't get the eventhub source example to start returning messages until I went into the portal and created the consumer group ID I wanted to use. I think we need to add a step here to do that.

File: gr-azure-software-radio/python/eventhub_source.py
Comment (pomeroy3) [reply gr-azure-software-radio/python/eventhub_source.py:98]: Ah and the eventhub_comsumer.close() shuts down the self.eventhub_consumer.receive() call. This is tidy, I like it. 

I never knew self.eventhub_consumer.receive would keep running across multiple batches of messages, I always thought it would return after it got to the end of the messages that were already on the partition. This is cool, nicely done.

**Review by pomeroy3 — CHANGES_REQUESTED** [review]
Looks great. I think the instructions for the examples are missing a step that tells people to create a consumer group in the Azure portal if they aren't using the default group, and we may need a similar step for the integration tests. (I haven't confirmed that one)

**Review by pomeroy3 — APPROVED** [review]
Approving after remaining readme update is merged in

**Pull request merged**
